src/parsers/manifestjson.js

```diff
@@ -1269,6 +1267,24 @@ export default class ManifestJSONParser extends JSONParser {
     }
   }
 
+  validateDataCollectionPermissions(permissions) {
+    if (!permissions) {
+      this.collector.addNotice(messages.MISSING_DATA_COLLECTION_PERMISSIONS);
+      return;
+    }
+
+    const { required } = permissions;
+    const requiredPermissions = Array.isArray(required) ? required : [];
+
+    if (
+      requiredPermissions.includes('none') &&
+      requiredPermissions.length > 1
+    ) {
+      this.collector.addError(messages.NONE_DATA_COLLECTION_IS_EXCLUSIVE);
+      this.isValid = false;
+    }
```

Comment:
thought: @willdurand should we also report a linting error if there are optional permissions but without any required data collection set in the manifest?

(if I'm not mistaken in that case we still want `required` to be set, either to `["none"]` or to a set of required data collection permissions.